### DISCUSSION TITLE
Errors when trying to compile the code on vs code

### DISCUSSION THREAD

**Discussion by tunank** [Cyfrin/foundry-full-course-cu#1691] [General]
Hi there, when I try tof forge build my code on Vs Code, it shows that
```
[⠊] Compiling...
[⠘] Installing Solc version 0.8.19
[⠃] Failed to install Solc 0.8.19: error decoding response body
Error: 
installation failed: error decoding response body
```
So I tried to down load the solc 0.8.19 externally. The download was successful but when I try to forge build again. The error still occurs. Anyone has any idea ? Thank you!
<img width="541" alt="Screenshot 2024-05-07 at 10 23 14 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/140475824/682b44cf-0fdb-48a3-acba-60afbcc528d1">

**Comment by EngrPips** [comment]
Hello @tunank, This is resolved already. Right?

**Comment by ACaiSec** [comment]
I think I face the same question:
```shell
MacBook-Pro project % forge build
Error: 
installation failed: error decoding response body
```
I have used the VPN.

**Reply by tunank** [reply]
Hi, try to turn off your VPN, restart the vscode and try the following code 
````
sudo npm uninstall -g solc
````
Then use the below command 
````
npm config delete proxy 
npm config delete https-proxy
````
````
node --trace-deprecation app.js
````
````
npm install -g npm@latest
````
````
export SOLC_BINARY_PATH=/path/to/solc
````

**Reply by EngrPips** [reply]
Hello @ACaiSec, Please try out @tunank suggestion and let us know if it fixes your problem.

**Reply by ACaiSec** [reply]
An error occurred when I executed the command `node --trace-deprecation app.js`. I would like to add that my project is not the same as @tunank , but a similar error occurred.
``` 
% node --trace-deprecation app.js
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module '/<path...>/project/app.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.17.1
```

**Reply by EngrPips** [reply]
Hello @ACaiSec, can you ask an AI buddy what this error is and what the possible solution is? Please post any solution that works here so it can serve as a reference for any other person who runs into the same issue.

**Comment by boycovskiykurt** [comment]
i am also met with this problem, but my version is 0.8.30
![111](https://github.com/user-attachments/assets/7f388057-a550-4690-82de-d25f948d6ef7)

**Reply by boycovskiykurt** [reply]
try to check solc version by that ` solcjs --version`

i find that solc already installed on my machine
![image](https://github.com/user-attachments/assets/53dc4b0f-a191-4f58-ac97-20cee5ecd6fe)

**Reply by EngrPips** [reply]
This could be an internet connection problem. You may also retry using a different solidity version.
